Sources/WhisperKit/Core/EnergyVAD.swift

```diff
@@ -0,0 +1,76 @@
+//  For licensing see accompanying LICENSE.md file.
+//  Copyright © 2024 Argmax, Inc. All rights reserved.
+
+import Foundation
+import Accelerate
+
+/// Voice activity detection based on energy threshold
+final class EnergyVAD {
+    var sampleRate: Int
+    var frameLength: Int
+    var frameShift: Int
+    var energyThreshold: Float
+
+    /// Initialize a new EnergyVAD instance
+    /// - Parameters:
+    ///   - sampleRate: audio sample rate
+    ///   - frameLength: frame length in milliseconds
+    ///   - frameShift: frame shift in milliseconds
+    ///   - energyThreshold: minimal energy threshold
+    init(
+        sampleRate: Int = 16000,
+        frameLength: Int = 25,
+        frameShift: Int = 20,
+        energyThreshold: Float = 0.05
+    ) {
+        self.sampleRate = sampleRate
+        // Compute frame length and frame shift in number of samples (not milliseconds)
+        self.frameLength = frameLength * sampleRate / 1000
+        self.frameShift = frameShift * sampleRate / 1000
+        self.energyThreshold = energyThreshold
+    }
+
+    func voiceActivity(in waveform: [Float]) -> [Bool] {
+        let count = (waveform.count - frameLength + frameShift) / frameShift
+        if count < 0 {
+            return []
+        }
+        return (0..<count).map { vDSP.sumOfSquares(waveform[$0 * frameShift..<($0 * frameShift + frameLength)]) > energyThreshold }
```

Comment:
I'd pull this into a helper function so it's a little easier to read. There is a similar section of code doing this that should move over https://github.com/argmaxinc/WhisperKit/blob/d6f50da1dc0b69fc53bb7d4a14f3972cc1353174/Sources/WhisperKit/Core/AudioProcessor.swift#L287-L321